### PR TITLE
Grab user ID and Skills from new employee_info property

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
HomePage positions now using new grade and skills property in employee_info. SkillCodeList component now excluding () from around skill-code if no skill-description available.

### CODE REVIEW

File: src/Components/ProfileDashboard/UserProfile/UserProfileGeneralInformation/UserProfileGeneralInformation.jsx [hunk 1]
```diff
@@ -22,18 +22,26 @@ const UserProfileGeneralInformation = ({ userProfile, showEditLink, useGroup, is
...
-      content={`Grade: ${get(userProfile, 'grade') || NO_GRADE}`}
+      content={`Grade: ${userGrade}`}
       className={infoDataPointClassName}
     />)
     :
-    (<StaticDevContent>
+    (<div>
+      <StaticDevContent>
+        <InformationDataPoint
+          content="Generalist •"
```
Comment (elizabeth-jimenez) [inline src/Components/ProfileDashboard/UserProfile/UserProfileGeneralInformation/UserProfileGeneralInformation.jsx:36]: @mjoyce91 wasn't sure what to make this look like, but this is ugly. Not sure what 'Generalist' usually takes the place of?

**Comment by elizabeth-jimenez** [comment]
@mjoyce91 @scott062 could not get one test to pass :( `src/Containers/Dashboard/Dashboard.test.jsx`

**Comment by mjoyce91** [comment]
@elizabeth-jimenez - what have you tried?

```
 Warning: Failed prop type: Invalid prop `skillCodes[0]` of type `number` supplied to `SkillCodeList`, expected `object`.
 
        in SkillCodeList (at UserProfileGeneralInformation.jsx:61)
```

**Comment by elizabeth-jimenez** [comment]
> @elizabeth-jimenez - what have you tried?
> 
> ```
>  Warning: Failed prop type: Invalid prop `skillCodes[0]` of type `number` supplied to `SkillCodeList`, expected `object`.
>  
>         in SkillCodeList (at UserProfileGeneralInformation.jsx:61)
> ```

I tried changing from `userProfile.employee_info.skills` to `userProfile.skills` hoping that would fix it. The problem is that in some places userProfile is now using the new data format `employee_info`, but not all places. This is the one where I wanted to print out the userProfile so I could at least see what I was dealing with, but couldn't find it lol

**Comment by mjoyce91** [comment]
@elizabeth-jimenez double check how you're passing the skills in to personal dashboard. When I go to the public profile for Jenny or Tarek, I can see their skills listed. But when I sign in as them and go their personal dashboard, I don't see the skills listed. If it makes sense to, feel free to change how the responses are structured so that there isn't variation between personal vs public dashboard.

**Comment by elizabeth-jimenez** [comment]
> @elizabeth-jimenez double check how you're passing the skills in to personal dashboard. When I go to the public profile for Jenny or Tarek, I can see their skills listed. But when I sign in as them and go their personal dashboard, I don't see the skills listed. If it makes sense to, feel free to change how the responses are structured so that there isn't variation between personal vs public dashboard.

ok will do!

**Comment by mjoyce91** [comment]
@elizabeth-jimenez one thing I noticed in the snapshots was that the skills array was coming through as a simple array of numbers, i.e. `[52, 4]` instead of an array of objects with code/description. That would at least explain the failing test.

**Comment by elizabeth-jimenez** [comment]
> @elizabeth-jimenez one thing I noticed in the snapshots was that the skills array was coming through as a simple array of numbers, i.e. `[52, 4]` instead of an array of objects with code/description. That would at least explain the failing test.

yea, that one must still be using the old version. That's also why in some of the other tests i would get the `expected number got string`, bc now our `employee_info.skills[#].code` is a string, where it used to come through as a number hehe 😵 ☠️

File: src/Containers/HomePagePositionsContainer/HomePagePositionsContainer.jsx [hunk 2]
```diff
@@ -18,21 +18,20 @@ class HomePagePositionsContainer extends Component {
 
   componentDidMount() {
     // Don't try to pull positions until we've received the user's profile.
-    if (this.props.userProfile.id) {
-      this.props.homePagePositionsFetchData(this.props.userProfile.skills,
-        this.props.userProfile.grade);
+    if (this.props.userProfile.emp_id) {
```
Comment (mjoyce91) [inline src/Containers/HomePagePositionsContainer/HomePagePositionsContainer.jsx:21]: I think you'll actually want to keep this as

```jsx
if (this.props.userProfile.id) {`
```

The reason being is that contractors don't have an emp_id, and I'm pretty sure that this section would just endlessly show a spinner if you kept this check.

File: src/Containers/HomePagePositionsContainer/HomePagePositionsContainer.jsx [hunk 3]
```diff
@@ -18,21 +18,20 @@ class HomePagePositionsContainer extends Component {
...
   }
 
   UNSAFE_componentWillReceiveProps(nextProps) {
     // Once we have a valid user profile, fetch the positions, but only
     // once. We'll set hasFetched to true to keep track.
-    if (nextProps.userProfile.id && !this.state.hasFetched && !this.props.homePagePositionsIsLoading
-      && !nextProps.homePagePositionsIsLoading) {
-      this.props.homePagePositionsFetchData(nextProps.userProfile.skills,
-        nextProps.userProfile.grade);
+    if (nextProps.userProfile.emp_id && !this.state.hasFetched
```
Comment (mjoyce91) [inline src/Containers/HomePagePositionsContainer/HomePagePositionsContainer.jsx:30]: Same comment as above about id vs emp_id

File: src/Containers/HomePagePositionsContainer/HomePagePositionsContainer.jsx (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/Containers/HomePagePositionsContainer/HomePagePositionsContainer.jsx]: Ah, good to know, thanks! Updated!

**Comment by mjoyce91** [comment]
@elizabeth-jimenez take a look at failing test and we should be good to go

**Pull request merged**